packages/x-charts/src/ChartsGrid/ChartsGrid.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import composeClasses from '@mui/utils/composeClasses';
+import { styled } from '@mui/material/styles';
+
+import { CartesianContext } from '../context/CartesianContextProvider';
+import useTicks from '../hooks/useTicks';
+import { ChartsGridClasses, getGridUtilityClass, gridClasses } from './gridClasses';
+
+export const GridRoot = styled('g', {
+  name: 'MuiChartsGrid',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})(({ theme }) => ({
+  [`& .${gridClasses.line}`]: {
+    stroke: (theme.vars || theme).palette.divider,
+    shapeRendering: 'crispEdges',
+    strokeWidth: 1,
+  },
+}));
+
+const useUtilityClasses = ({ classes }: ChartsGridProps) => {
+  const slots = {
+    root: ['root'],
+    verticalLine: ['line', 'verticalLine'],
+    horizontalLine: ['line', 'horizontalLine'],
+  };
+
+  return composeClasses(slots, getGridUtilityClass, classes);
+};
+
+export interface ChartsGridProps {
+  /**
+   * Displays vertical grid.
+   */
+  vertical?: boolean;
```

Comment:
Have you considered going for props less reliant on cartesian coordinates?
For example, if we wanted grid on "Pie" or [something similar](https://docs.telerik.com/devtools/winforms/controls/chartview/features/chart-grid#polar-grid), having `horizontal` and `vertical` might be a bit of a bottleneck. 🙈

Reply:
That naming conflict might be negligible compare to supporting the `xAxis` and `yAxis` props ;)

I think the grid will end-up with a typing of the form `CartesianGridProps | PolarGridProps`.

Since no chart use both polar and cartesian at the same time, it feels better to have a config for each with naming that make sens.

For example:

- Echarts has `radiusAxis` and `angleAxis` in addition of the `xAxis` and `yAxis`
- telerik has `PolarGrid`and `CartesianGrid` with explicit properties such as `DrawHorizontalFills` or `DrawRadialStripes`